Documentation/ceph-object-bucket-claim.md

```diff
@@ -0,0 +1,241 @@
+  BUCKET_PORT: 80
+  BUCKET_NAME: photo-booth-62PrQ
+  BUCKET_REGION: us-west-1
+```
+
+##### App Pod
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app-pod
+  namespace: dev-user
+spec:
+  containers:
+  - name: mycontainer
+    image: redis
+    envFrom: [1]
+    - configMapRef:
+        name: ceph-bucket [2]
+    - secretRef:
+        name: ceph-bucket [3]
+```
+1. use `env:` if mapping of the defined key names to the env var names used by the app is needed.
+1. makes available to the pod as env variables: `BUCKET_HOST`, `BUCKET_PORT`, `BUCKET_NAME`
+1. makes available to the pod as env variables: `ACCESS_KEY_ID`, `SECRET_ACCESS_KEY`
+
+##### Generated OB Custom Resource
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucket
+metadata:
+  creationTimestamp: "2019-10-02T01:35:36Z"
+  finalizers:
+  - objectbucket.io/finalizer
+  generation: 1
+  labels:
+    bucket-provisioner: aws-s3.io-bucket
+  name: obc-default-ceph-bucket
+  resourceVersion: "7761"
+  selfLink: /apis/objectbucket.io/v1alpha1/objectbuckets/obc-s3-provisioner-myobc
+  uid: eeb781ae-e4b4-11e9-ad72-982e45ddf979
+spec:
+  additionalState:
+    ARN: arn:aws:iam::939345161466:policy/photo-booth-62PrQ
+    UserName: photo-booth-62PrQ
+  claimRef:
+    apiVersion: objectbucket.io/v1alpha1
+    kind: ObjectBucketClaim
+    name: obc-default-ceph-bucket
+    namespace: default
+    uid: 6506150e-e4aa-11e9-ad72-982e45ddf979
+  endpoint:
+    additionalConfig:
+      "": ""
+    bucketHost: http://my-store
+    bucketName: photo-booth-62PrQ
+    bucketPort: 80
+    region: us-west-1
+    subRegion: ""
+  reclaimPolicy: Delete
+  storageClassName: rook-ceph-bucket
+status:
+  phase: Bound
+```
+
+##### StorageClass
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: rook-ceph-bucket
+  labels:
+    aws-s3/object [1]
+provisioner: ceph.rook.io/bucket [2]
+parameters: [3]
+  objectStoreName: my-store
+  objectStoreNamespace: rook-ceph
+  region: us-west-1
+  bucketName: ceph-bucket [4]
+reclaimPolicy: Delete [5]
+```
+1. `label`(optional) here associates this `StorageClass` to a specific provisioner.
+1. `provisioner` responsible for handling `OBCs` referencing this `StorageClass`.
+1. **all** `parameter` keys and values are specific to a provisioner, are optional, and are not validated by the `StorageClass` API.
+Fields to consider are `object-store endpoint`, `version`, etc.
+1. `bucketName` is required for access to existing buckets but is omitted when provisioning new buckets.
+Unlike greenfield provisioning, the brownfield bucket name appears in the `StorageClass`, not the `OBC`.
+1. each provisioner decides how to treat the `reclaimPolicy` when an `OBC` is deleted for the bucket. See explaination as [specified in Kubernetes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#retain)
++ _Delete_ = (typically) physically delete the bucket.
++ _Retain_ = (typically) do not physically delete the bucket.
```

Comment:
Drop "(typically)".  Rook-Ceph handles each in a specific way.

Reply:
done